### PR TITLE
feat: support forcing persistence of partition irrespective of arrival time

### PR DESCRIPTION
**Pull request by tustvold**
I originally tried to roll addressing #3247 into this, but it got really complicated. This just adds the ability to force persistence, which I need in order to properly test #3224 

Edit: I found an alternative way to test #3224 but still think this change makes sense

### CODE REVIEW

File: generated_types/protos/influxdata/iox/management/v1/service.proto [hunk 1]
```diff
@@ -402,6 +402,9 @@ message PersistPartitionRequest {
 
   // the table name
   string table_name = 3;
+
+  // persist all data for partition irrespective of arrival time
```
Comment (alamb) [inline generated_types/protos/influxdata/iox/management/v1/service.proto:406]: I agree this makes sense

File: influxdb_iox/src/commands/database/partition.rs [hunk 2]
```diff
@@ -229,7 +229,7 @@ pub async fn command(connection: Connection, config: Config) -> Result<()> {
             } = persist;
 
             client
-                .persist_partition(db_name, table_name, partition_key)
```
Comment (alamb) [inline influxdb_iox/src/commands/database/partition.rs:232]: Is there any value exposing a `--force` cli flag too?

**Review by alamb — APPROVED** [review]
makes sense to me

File: influxdb_iox/src/commands/database/partition.rs (hunk 2 above)
Comment (tustvold) [reply influxdb_iox/src/commands/database/partition.rs:232]: Done

**Pull request merged**